src/pip/_internal/wheel.py

```diff
@@ -791,24 +791,33 @@ def should_use_ephemeral_cache(
     if req.constraint:
         # never build requirements that are merely constraints
         return None
+
     if req.is_wheel:
-        if not should_unpack:
+        if must_build:
             logger.info(
                 'Skipping %s, due to already being wheel.', req.name,
             )
+        # never build a requirement that is already a wheel
+        return None
+
+    if req.editable:
+        # if must_build an editable, build to ephem cache,
+        # else do not build as install will egglink to it
+        if must_build:
+            return True
         return None
-    if not should_unpack:
-        # i.e. pip wheel, not pip install;
-        # return False, knowing that the caller will never cache
-        # in this case anyway, so this return merely means "build it".
-        # TODO improve this behavior
-        return False
 
-    if req.editable or not req.source_dir:
+    if not req.source_dir:
+        # TODO explain what no source_dir means
+        if must_build:
+            return True
         return None
 
     if "binary" not in format_control.get_allowed_formats(
             canonicalize_name(req.name)):
+        if must_build:
+            # --no-binary has no effect as a pip wheel option
+            return True
```

Comment:
Instead of returning `True` here, why wouldn't you instead want to skip this clause and continue on (so make the if check `if not must_build and "binary" not in format_control ...`)?

Reply:
I found it nice to have the symmetry with other cases, and have a place to put that little comment about `pip wheel --no-binary` which I thought useful.

Reply:
What I'm saying is that it seems like this can cause the wrong return value. If there is a condition later on that can result in a return value of `False`, the way the code is here short-circuits that (and in particular means that `--no-binary` _is_ affecting the return value.

Reply:
In any case, if you do the `should_build()` approach, this issue will go away because the code will be structured differently. (It will wind up being equivalent to what I was suggesting you do here.)